### ISSUE TITLE
[BUG] update.sh doesn't detect node version correctly

### ISSUE THREAD

**Issue by thargy** [bitfocus/companion#2202]
### Is this a bug in companion itself or a module?

- [X] I believe this to be a bug in companion

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I need to do some [enhancements to a module](https://github.com/bitfocus/companion-module-thelightingcontroller/issues/6) I published 3 years ago, so trying to set up a new dev environment using latest WSL2; based on a user request.

### Steps To Reproduce

Followed [Developing on Windows Subsystem for Linux](https://github.com/bitfocus/companion/wiki/Developing-on-Windows-Subsystem-for-Linux) from the Wiki.  This has a number of problems; including the reference to 'libgconf2-4' which I think should be 'libgconf-2-4', and a lot of missing steps like installing yarn, etc.  However, I managed to install everything and checkout the companion repository.

I also installed `nvm` (note it uses '14') and `python` (note this command also no longer works, as you need to specify `python3` and `python-is-python3`.

Running `yarn update`, gives:
```
00h00m00s 0/0: : ERROR: [Errno 2] No such file or directory: 'update'
```
whilst in the companion root folder.

So, I switched to [the Module Development tutorial](https://github.com/bitfocus/companion/wiki/Module-Development); and tried to run ` ./tools/update.sh`

That gave me this:
```
Already up to date.
Check Node version
Cannot find module 'lru-cache'
Require stack:
- /usr/share/nodejs/semver/classes/range.js
- /usr/share/nodejs/semver/classes/comparator.js
- /usr/share/nodejs/semver/index.js
- /usr/share/nodejs/semver/bin/semver.js
Found v14.19.0
The installed version of NodeJS is not supported, ">=14.18 <15" is required.
It is recommended that you update NodeJS (the same way you installed it).
```

I used nvm to try installing different versions, update the default alias, etc. here's the result of `nvm ls`:

```
        v14.9.0
->     v14.19.0
       v14.19.3
       v14.21.1
         system
default -> 14.19.0 (-> v14.19.0)
node -> stable (-> v14.21.1) (default)
stable -> 14.21 (-> v14.21.1) (default)
iojs -> N/A (default)
unstable -> N/A (default)
lts/* -> lts/hydrogen (-> N/A)
lts/argon -> v4.9.1 (-> N/A)
lts/boron -> v6.17.1 (-> N/A)
lts/carbon -> v8.17.0 (-> N/A)
lts/dubnium -> v10.24.1 (-> N/A)
lts/erbium -> v12.22.12 (-> N/A)
lts/fermium -> v14.21.1
lts/gallium -> v16.18.1 (-> N/A)
lts/hydrogen -> v18.12.1 (-> N/A)
```

`14.9.0` matches `>=14.18 <15` based on my understanding of semver?

This may be related to #1531?

Sadly, having wasted over 2 hours and not gotten anywhere, I'm now stuck and have no time to support the user.

### Expected Behavior

Should be able to follow tutorials to setup a clean development environment so I can quickly support users that have queries about legacy modules.

### Environment (please complete the following information)

```markdown
- OS: Windows 11, using WSL 2 (latest version) and Ubuntu. Clean install
- Browser: N/A (can't build)
- Companion Version: Clean checkout (2.4.0, HEAD, cd0b68c6)
```

**Comment by thargy** [comment]
Running `npx semver --range ">=14.18 <15" "v14.9.0"` gives:

```
Cannot find module 'lru-cache'
Require stack:
- /usr/share/nodejs/semver/classes/range.js
- /usr/share/nodejs/semver/classes/comparator.js
- /usr/share/nodejs/semver/index.js
- /usr/share/nodejs/semver/bin/semver.js
```

So this may be something to do with how `npx` is set up?

**Comment by thargy** [comment]
Running `npm install -g semver` got me around that issue, now running `update.sh` gets as far as:

```
Already up to date.
Check Node version
Found v14.19.0
Node version is OK
Core
Usage: yarn [options]

yarn: error: no such option: --frozen-lockfile
```

**Comment by Julusian** [comment]
My first question is do you need the update quickly and in a stable-ish version of companion?
We will only be accepting updates to modules which have been updated to the 3.0 module format https://bitfocusio.slack.com/archives/CFG7HAN5N/p1669405914683829

But yes this is probably related to #1531. Something to do with using both `n` and `npx` gets something really broken, it could be the same for other node version managers, but strangely I've never had an issue when using `nvm` or `fnm` on linux..

I am hoping that this can be resolved by instead updating the module for companion 3.0 (which changes how your machine should be setup, I dont think there is any need for wsl), but if not then Ill try and think of some ideas

**Comment by thargy** [comment]
Hi @Julusian,

Thanks for the response, I just saw this on Slack too and started researching the update process.  You'll see from [this comment](https://github.com/bitfocus/companion/issues/2202#issuecomment-1328036665) that I got further by installing `semver` and `yarn` properly, but was still stuck.

I'll liaise with the user, as I've given them a temp workaround. Instead, I'll look at instead doing the update for v.3.
